python/lsst/ctrl/mpexec/dotTools.py

```diff
@@ -244,7 +244,13 @@ def expand_dimensions(dimensions):
                 dimensions = expand_dimensions(attr.dimensions)
                 _renderDSTypeNode(attr.name, dimensions, file)
                 allDatasets.add(attr.name)
+            nodeName, component = DatasetType.splitDatasetTypeName(attr.name)
             _renderEdge(attr.name, taskNodeName, file)
+            # connect component dataset types to the composite type that
+            # produced it
+            if component is not None and (nodeName, attr.name) not in allDatasets:
+                _renderEdge(nodeName, attr.name, file)
```

Comment:
Should this also need:

    if nodeName not in allDatasets:
        _renderDSTypeNode(nodeName, ...)

(in case a component is a pre-existing input and not an intermediate)?

It may be a nicer representation if a component could be rendered as a box inside a dataset type node (e.g. using record-based graphviz nodes), but that is probably out of scope for this ticket.

Reply:
I am not sure what you mean with your first point, the rendering of the DSType node is handled above, and prerequisites are handled in a separate loop.

I thought about your latter point, but decided it was not worth re-working so much for a quicker fix that seems to do its job well enough.

Reply:
ah yeah, I see what you are saying, good point, ill fix that up